lib/src/main/java/com/basistheory/android/view/CardNumberElement.kt

```diff
@@ -43,14 +43,31 @@ class CardNumberElement @JvmOverloads constructor(
         isEmpty: Boolean,
         isValid: Boolean
     ): ChangeEvent {
-        val eventDetails = this.cardMetadata?.brand?.let { brand ->
-            mutableListOf(
+        val eventDetails = mutableListOf<EventDetails>()
+
+        if (this.cardMetadata?.brand != null)
+            eventDetails.add(
+                EventDetails(
+                    EventDetails.CardBrand,
+                    this.cardMetadata?.brand ?: ""
+                )
+            )
+
+        if (value != null && cardMetadata?.isComplete == true) {
+            eventDetails.add(
+                EventDetails(
+                    EventDetails.Bin,
+                    value.take(6)
+                )
+            )
+
+            eventDetails.add(
                 EventDetails(
-                    "cardBrand",
-                    brand
+                    EventDetails.Last4,
```

Comment:
why would last 4 be added even if it is not complete?

Reply:
What do you mean? This is only adding the last4 if the element is complete.